portal/src/main/webapp/js/src/study-view/component/ScatterPlots.js

```diff
@@ -1152,47 +1168,25 @@ var ScatterPlots = function() {
                     }
                 }
             }
-            
+
             for(var j=0 ; j< _datumArr.length ; j++){
-                if(_datumArr[j].fill !== 'red') {
-                    for(var i=0 ; i< dataCopyL ; i++){
-                        if(_datumArr[j].case_id === dataCopy[i].case_id){
-                            _tmpDataArr.unshift(dataCopy[i]);
-                            break;
-                        }
-                    }
-                }else {
-                    for(var i=0 ; i< dataCopyL ; i++){
-                        if(_datumArr[j].case_id === dataCopy[i].case_id ){
-                            _tmpDataArr.push(dataCopy[i]);
-                            break;
-                        }
-                    }
+                if(_datumArr[j].fill === 'red') {
+                    highlightCases[_datumArr[j].case_id] = j;
                 }
-                _caseIdList.push(_datumArr[j].case_id);
+                caseIdList[_datumArr[j].case_id] = j;
             }
 
-            dataCopy = _tmpDataArr;
-            drawPlots(dataCopy);
-            addQtips();
-
-            elem.dotsGroup.selectAll("path").each(function(d) {
-                var _index = _caseIdList.indexOf(d.case_id);
-                if (_index !== -1) {
-                    $(this).attr("fill", _datumArr[_index].fill);
-                    $(this).attr("stroke", _datumArr[_index].stroke);
-//                    $(this).attr("opacity", _datumArr[_index].opacity);
-                    $(this).attr("d", d3.svg.symbol().size(_datumArr[_index].size).type(style.shape));
-                    $(this).attr("stroke-width", _datumArr[_index].strokeWidth);
-                    if(_datumArr[_index].fill === style.fill && _datumArr[_index].stroke === 'red'){
-                        $(this).attr("clicked", 'clicked');
-                    }else if(_datumArr[_index].fill === 'red' && _datumArr[_index].stroke === 'red'){
-                        $(this).attr("clicked", 'shiftClicked');
-                    }else if(_datumArr[_index].fill === 'red' && _datumArr[_index].stroke === style.stroke){
-                        $(this).attr("clicked", 'both');
-                    }
+            for(var i=0 ; i< dataCopyL ; i++){
+                var _case_id = dataCopy[i].case_id;
+                if(highlightCases.hasOwnProperty(_case_id)) {
```

Comment:
You are right!! Check this out: http://jsperf.com/key-exist-check/3 I will start to use your way from now on.